java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -0,0 +1,476 @@
+        break;
+      case NEW_ARRAY:
+        NewArrayTree newArrayTree = (NewArrayTree) tree;
+        programState = ProgramState.unstack(programState, newArrayTree.initializers().size()).a;
+        SymbolicValue svNewArray = constraintManager.createSymbolicValue(newArrayTree);
+        programState = ProgramState.stackValue(programState, svNewArray);
+        programState = svNewArray.setSingleConstraint(programState, NullConstraint.NOT_NULL);
+        break;
+      case NEW_CLASS:
+        NewClassTree newClassTree = (NewClassTree) tree;
+        programState = ProgramState.unstack(programState, newClassTree.arguments().size()).a;
+        SymbolicValue svNewClass = constraintManager.createSymbolicValue(newClassTree);
+        programState = ProgramState.stackValue(programState, svNewClass);
+        programState = svNewClass.setSingleConstraint(programState, NullConstraint.NOT_NULL);
+        break;
+      case MULTIPLY:
+      case DIVIDE:
+      case REMAINDER:
+      case PLUS:
+      case MINUS:
+      case LEFT_SHIFT:
+      case RIGHT_SHIFT:
+      case UNSIGNED_RIGHT_SHIFT:
+      case AND:
+      case XOR:
+      case OR:
+      case GREATER_THAN:
+      case GREATER_THAN_OR_EQUAL_TO:
+      case LESS_THAN:
+      case LESS_THAN_OR_EQUAL_TO:
+      case EQUAL_TO:
+      case NOT_EQUAL_TO:
+        // Consume two and produce one SV.
+        Pair<ProgramState, List<SymbolicValue>> unstackBinary = ProgramState.unstack(programState, 2);
+        programState = unstackBinary.a;
+        SymbolicValue symbolicValue = constraintManager.createSymbolicValue(tree);
+        symbolicValue.computedFrom(unstackBinary.b);
+        programState = ProgramState.stackValue(programState, symbolicValue);
+        break;
+      case POSTFIX_INCREMENT:
+      case POSTFIX_DECREMENT:
+      case PREFIX_INCREMENT:
+      case PREFIX_DECREMENT:
+      case UNARY_MINUS:
+      case UNARY_PLUS:
+      case BITWISE_COMPLEMENT:
+      case LOGICAL_COMPLEMENT:
+      case INSTANCE_OF:
+        // consume one and produce one
+        Pair<ProgramState, List<SymbolicValue>> unstackUnary = ProgramState.unstack(programState, 1);
+        programState = unstackUnary.a;
+        SymbolicValue unarySymbolicValue = constraintManager.createSymbolicValue(tree);
+        unarySymbolicValue.computedFrom(unstackUnary.b);
+        programState = ProgramState.stackValue(programState, unarySymbolicValue);
+        break;
+      case IDENTIFIER:
+        Symbol symbol = ((IdentifierTree) tree).symbol();
+        SymbolicValue value = programState.values.get(symbol);
+        if (value == null) {
+          value = constraintManager.createSymbolicValue(tree);
+          programState = ProgramState.put(programState, symbol, value);
+        }
+        programState = ProgramState.stackValue(programState, value);
+        break;
+      case MEMBER_SELECT:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 8 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1151)